lib/_stream_readable.js

```diff
@@ -275,10 +275,11 @@ function readableAddChunk(stream, chunk, encoding, addToFront) {
   debug('readableAddChunk', chunk);
   const state = stream._readableState;
 
-  let skipChunkCheck;
-
+  let err;
   if (!state.objectMode) {
-    if (typeof chunk === 'string') {
+    if (chunk instanceof Buffer) {
+      encoding = '';
+    } else if (typeof chunk === 'string') {
```

Comment:
Nit: it's probably a good idea to switch the order of the instanceof and typeof check. `typeof` should be faster and probably the more common case.